### PR TITLE
16.0

### PR DESCRIPTION
**Pull request by arnis-putnins**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by arnis-putnins** [comment]
Hi, can you tell what's wrong with this: https://runbot.odoo.com/runbot/build/24246399

**Comment by antoine162** [comment]
Hi @arnis-putnins this is an error which is affecting all new localizations at the moment, we have not solved it yet but are planning to solve it soon :smile: 
The tests will be re-run next time you push, so that error should go away if you push again in a few days.

File: addons/l10n_lv/data/account_tax_template_data.xml [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+            (0,0, {
+              'repartition_type': 'tax',
+              'account_id': ref('a5721'),
+              'tag_ids': [ref('VAT_41P_tag')],
+            }),
+        ]" />
+  </record>
+
+  <!--  VAT, VAT report r411 reversa preces un pakalpojumi -->
+  <record id="account_tax_template_sales_VAT_411S_tag" model="account.tax.template">
```
Comment (antoine162) [inline addons/l10n_lv/data/account_tax_template_data.xml:41]:
```suggestion
  <record id="account_tax_template_sales_VAT_411S" model="account.tax.template">
```
Better rename this to avoid confusion.

File: addons/l10n_lv/data/account_tax_template_data.xml [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+            }),
+        ]" />
+  </record>
+
+  <!--  VAT, VAT report PVN2 services-->
+  <record id="account_tax_template_sales_VAT_2P_tag" model="account.tax.template">
+    <field name="chart_template_id" ref="account_chart_template_latvia" />
+    <field name="name">VAT2 Services</field>
+    <field name="description">W/O VAT (EU)</field>
+    <field name="type_tax_use">sale</field>
```
Comment (antoine162) [inline addons/l10n_lv/data/account_tax_template_data.xml:410]: Not a big deal, but could we please define all sales taxes first, then the purchase taxes?

File: addons/l10n_lv/data/account_tax_template_data.xml [hunk 3]
```diff
@@ -0,0 +1,557 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo noupdate="1">
+
+  <!-- Sales -->
+  <!-- 21% VAT, VAT report r41 + r51 + PVN 1.1. -->
+  <record id="account_tax_template_sales_21_vat41S" model="account.tax.template">
```
Comment (antoine162) [inline addons/l10n_lv/data/account_tax_template_data.xml:6]:
```suggestion
  <record id="account_tax_template_sales_21_vat41S" model="account.tax.template">
    <field name="sequence">100</field>
```
Could we please add the `sequence` field to each tax, which is an increasing number? (e.g. 100, 110, 120 etc.)

File: addons/l10n_lv/data/res_bank_data_lt.xml [hunk 4]
```diff
@@ -0,0 +1,122 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo noupdate="1">
+
+    <record id="res_bank_CBSBLT26XXX" model="res.bank">
```
Comment (antoine162) [inline addons/l10n_lv/data/res_bank_data_lt.xml:4]: Thanks so much for this. I see this is an update of the banks provided in `l10n_lt`.
Could we split this off into a separate PR that just modifies `l10n_lt/data/res_bank_data.xml`?

File: addons/l10n_lv/data/account_account_tag_data.xml [hunk 5]
```diff
@@ -0,0 +1,1504 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+
+  <record id="account_account_tag_a" model="account.account.tag">
```
Comment (antoine162) [inline addons/l10n_lv/data/account_account_tag_data.xml:4]: The data for the Balance Sheet and Profit and Loss is worth keeping somewhere, since it will be useful for creating those reports on top of on the new reports engine.

File: addons/l10n_lv/data/account_account_tag_data.xml [hunk 6]
```diff
@@ -0,0 +1,1504 @@
...
+
+  <record id="account_account_tag_pza2_290" model="account.account.tag">
+    <field name="name">PZA2.290. Profit or loss for the current year</field>
+    <field name="applicability">accounts</field>
+    <field name="color" eval="8"/>
+  </record>
+
+  <!-- Tags for taxes -->
+
+  <!-- VAT report 1.1. types -->
```
Comment (antoine162) [inline addons/l10n_lv/data/account_account_tag_data.xml:1017]: The account tags for the VAT report should be defined directly in the report definition, in the `account.report.line`. 
See https://github.com/odoo/odoo/blob/b7232b14b7f2423241b24b4eef1300f1a1b425cf/addons/l10n_bg/data/tax_report.xml#L31-L34 for an example - this creates a tax tag with xmlid "l10n_bg_tax_report_11_tag" and which corresponds to the tax grid "11".

**Review by antoine162 — COMMENTED** [review]
Hi @arnis-putnins thank you so much for your work, I have reviewed and tested your PR and in general it seems to be of a very high standard. 

I have discussed with @chklop who is one of our product owners for the Accounting app. In terms of a minimum viable product (something that is worth merging into Odoo), we want to have 4 things working well:
- a chart of accounts 
- taxes and VAT report
- fiscal positions
- a demo company.

We are very nearly there. As I will explain below, there is just one main issue that needs to be addressed, which is the definition of the VAT report. 

Please find below an overview of the things that need changing. Once these are corrected, I would be happy to do any remaining small fixes myself, and merge this localization. 

**1. Chart of Accounts**
- The accounts are good. However, please remove the account tags corresponding to the Balance Sheet (as account tags are no longer used for the Balance Sheet since version 16 - more on that below), and if possible replace them with one of three tags used for the Cash Flow Statement (see inline comment). 

**2. Taxes and VAT Report**
- The taxes are good.
- For the VAT Report, I understand that you created account tags, since that was how the report worked before Odoo 16. :smile: (I note you also provided a Balance Sheet and a Profit and Loss report.)
- But from version 16 onwards, the reports use a new reports engine. You may have noticed that your report doesn't actually work if you go, for example, to Accounting > Reports > Tax Report.
- That is because the way of defining reports has changed.
- For example, this is how we define the VAT report in a localization that we did recently (Bulgaria). As you can see, the tax tags are defined directly in the report: https://github.com/odoo/odoo/blob/16.0/addons/l10n_bg/data/tax_report.xml
- Since we are keen to merge something as soon as possible, I suggest that we don't do the Balance Sheet and Profit and Loss right now. Keep the code you wrote, and feel free to make a PR for the Balance Sheet and Profit and Loss at a later time.
- The VAT Report however is absolutely necessary to have, and it needs to be written for the report engine in 16.0 for it to work.

**3. Fiscal Positions**
- The fiscal positions look good.
- Could we add a fiscal position for Intracom B2C? In this case the VAT needs to be paid directly in the consumer's country. It's not necessary to define any taxes (we expect users to define the appropriate taxes in this scenario) but having the fiscal position is nice. For an example, see https://github.com/odoo/odoo/blob/16.0/addons/l10n_bg/data/account_fiscal_position_template.xml#L13-L14

**4. Demo company**
- Looks good to me.
